main.go

```diff
@@ -184,6 +188,22 @@ func main() {
 		}
 	}
 
+	if res, err := outils.IsGVKDefined(schema.GroupVersionKind{
+		Group:   "cert-manager.io",
+		Version: "v1",
+		Kind:    "CertificateRequest",
+	}, discoveryClient); err == nil && res != nil {
```

Comment:
We need to differentiate between transient network errors and the CRD actually not being available.